build/arch.amd64.mk

```diff
@@ -0,0 +1,10 @@
+TARGET := x86_64-mimiker-elf
+GCC_ABIFLAGS :=
+CLANG_ABIFLAGS := -target x86_64-elf
+ELFTYPE := elf64-x86-64
+ELFARCH := i386:x86-64
+LDFLAGS += -z max-page-size=0x1000
+
+ifeq ($(KERNEL), 1)
+CFLAGS += -mno-red-zone
+endif
```

Comment:
A command has been added.

Reply:
Ok, and I still doesn't know what is a __real__ problem here and how that flag __fixes__ that problem.

Reply:
In kernel the privilege level is 0, and thus an interrupt handling doesn't switch the stack, therefore the current stack is used. A leaf functions are free to use a so called read zone: 
```
The 128-byte area beyond the location pointed to by %rsp is considered to be reserved and shall not be modified by signal or interrupt handlers. Therefore, functions may use this area for temporary data that is not needed across function calls. In particular, leaf functions may use this area for their entire stack frame, rather than adjusting the stack pointer in the prologue and epilogue. This area is known as the red zone.
```
The above flag ensures that GCC won't use red zones throughout the kernel (instead, data will be explicitly allocated on the stack).

Reply:
Much better explanation. Please add comment with that info or link to documentation.

Reply:
Done.